packages/@lwc/integration-not-karma/configs/plugins/serve-integration.js

```diff
@@ -65,27 +67,58 @@ const transform = async (ctx) => {
     const customLwcRollupPlugin = {
         ...defaultRollupPlugin,
         transform(src, id) {
-            const { apiVersion, nativeOnly } = parseConfig(src, id);
-
+            const { apiVersion, nativeOnly, featureFlag } = parseConfig(src, id);
             let transform;
             if (apiVersion) {
-                transform = createRollupPlugin(input, { apiVersion }).transform;
+                transform = createRollupPlugin(input, {
+                    apiVersion,
+                    featureFlag,
+                }).transform;
             } else if (nativeOnly) {
                 transform = createRollupPlugin(input, {
                     disableSyntheticShadowSupport: true,
+                    featureFlag,
                 }).transform;
             } else {
-                transform = defaultRollupPlugin.transform;
+                // Default path: if a featureFlag was provided via directive, pass it through
+                transform = featureFlag
+                    ? createRollupPlugin(input, { featureFlag }).transform
+                    : defaultRollupPlugin.transform;
             }
 
             return transform.call(this, src, id);
         },
     };
 
+    const featureFlagResolver = {
```

Comment:
```suggestion
    /**
     * Transforms `@salesforce/featureFlag/*` imports into modules that export a boolean.
     * `@salesforce/featureFlag/TEST_FLAG_ENABLED` exports true, all others export false.
     */
    const featureFlagResolver = {
```
There's too much magic in integration tests, so I always try to over-document.